src/components/AuthForm.js

```diff
@@ -512,6 +513,51 @@ export default class AuthForm extends PureComponent<
     }
   };
 
+  componentDidUpdate(prevProps: AuthFormProps, prevState: AuthFormState) {
+    const {
+      formData: { server: prevServer },
+    } = prevState;
+    const {
+      formData: { server: newServer },
+    } = this.state;
+
+    if (prevServer !== newServer) {
+      // Server changed, set the authType back to "anonymous"
+      return this.onChange({
+        ...this.state,
+        formData: { ...this.state.formData, authType: ANONYMOUS_AUTH },
+      });
+    }
+
```

Comment:
Please add a comment: `We will look for this server in the history to reuse the auth type...`

Reply:
Done that on line 526 and 527, thanks for the heads up